### PR TITLE
PAINTROID-629: Make bottom view in help dialog dynamically resize-able by implementing custom theme

### PR DESCRIPTION
**Pull request by msesko**
[PAINTROID 629](https://catrobat.atlassian.net/jira/software/c/projects/PAINTROID/boards/130?selectedIssue=PAINTROID-629): Make bottom view in help dialog dynamically resize-able by implementing custom theme

My solution: Expand colored area & change padding for better click-ability

image
image

## New Features and Enhancements
- Expand colored area & change padding for better clickability
## Refactorings and Bug Fixes
- Nothing new

#### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Paintroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#paintroid* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

**Comment by juliajulie95** [comment]
please link to atlassian, not jira

**Comment by msesko** [comment]
> please link to atlassian, not jira

done

**Review by Lenkomotive — CHANGES_REQUESTED** [review]
Hey! Instead of just filling in with the same color and "hiding" the problem, please try to find out why the bottom-sheet is not using the entire width.

**Comment by msesko** [comment]
> Hey! Instead of just filling in with the same color and "hiding" the problem, please try to find out why the bottom-sheet is not using the entire width.

I disagree. It is a UI problem, it was fixed since the user cannot spot a difference anyways. The simplest solution is the best imo.

Regardless I can do the following:

1. wrap the sheet with MediaQuery.removePadding -> did not work for me
2. build a custom bottomSheet and stack it on top of the current content -> introduces a bunch of other things to watch out for like overflows etc.

Is it really necessary to do this? To me changing the background seems the easiest solution. Any solution other than that I tried seems to ignore constraints created by Scaffold. There must surely be a reason why Scaffold does not allow the sheet to span the whole width (reachability)?

**Comment by Lenkomotive** [comment]
> > Hey! Instead of just filling in with the same color and "hiding" the problem, please try to find out why the bottom-sheet is not using the entire width.
> 
> I disagree. It is a UI problem, it was fixed since the user cannot spot a difference anyways. The simplest solution is the best imo.
> 
> Regardless I can do the following:
> 
> 1. wrap the sheet with MediaQuery.removePadding -> did not work for me
> 2. build a custom bottomSheet and stack it on top of the current content -> introduces a bunch of other things to watch out for like overflows etc.
> 
> Is it really necessary to do this? To me changing the background seems the easiest solution. Any solution other than that I tried seems to ignore constraints created by Scaffold. There must surely be a reason why Scaffold does not allow the sheet to span the whole width (reachability)?

It is not about just finding the simplest solutions, as this is a student project its totally ok to take time and dig deeper into some problems. Thats why I asked you to try to find the actual rootcause. I am aware that the user will see no difference and your solution is simple, yet it will still be unclear for us, why the bottomsheet behaves the way it behaves. 

I tried to research it myself and came up with folllowing: The constrains are set inside the `BottomSheetThemeData` which is inside the `ThemeData`.  Adjusting the constrains there allow us to use the entire width.  Since we have no specified own `ThemeData`, it would be indeed too much work for this ticket, but might be good to consider for the project, as we are facing other issues with BottomSheets.

So, 1 and 2 would have no affect anyway and changing the background is only a workaround imo.

[PAINTROID-703](https://catrobat.atlassian.net/browse/PAINTROID-703)  and [PAINTROID-455](https://catrobat.atlassian.net/browse/PAINTROID-455) are also facing some issues regarding the BottomSheet, maybe its worth to discuss how we can deal with BottomSheets in genreal in the project @bakicelebi @juliajulie95 @exland @bhav-khurana ?

**Comment by msesko** [comment]
> > > Hey! Instead of just filling in with the same color and "hiding" the problem, please try to find out why the bottom-sheet is not using the entire width.
> > 
> > 
> > I disagree. It is a UI problem, it was fixed since the user cannot spot a difference anyways. The simplest solution is the best imo.
> > Regardless I can do the following:
> > 
> > 1. wrap the sheet with MediaQuery.removePadding -> did not work for me
> > 2. build a custom bottomSheet and stack it on top of the current content -> introduces a bunch of other things to watch out for like overflows etc.
> > 
> > Is it really necessary to do this? To me changing the background seems the easiest solution. Any solution other than that I tried seems to ignore constraints created by Scaffold. There must surely be a reason why Scaffold does not allow the sheet to span the whole width (reachability)?
> 
> It is not about just finding the simplest solutions, as this is a student project its totally ok to take time and dig deeper into some problems. Thats why I asked you to try to find the actual rootcause. I am aware that the user will see no difference and your solution is simple, yet it will still be unclear for us, why the bottomsheet behaves the way it behaves.
> 
> I tried to research it myself and came up with folllowing: The constrains are set inside the `BottomSheetThemeData` which is inside the `ThemeData`. Adjusting the constrains there allow us to use the entire width. Since we have no specified own `ThemeData`, it would be indeed too much work for this ticket, but might be good to consider for the project, as we are facing other issues with BottomSheets.
> 
> So, 1 and 2 would have no affect anyway and changing the background is only a workaround imo.
> 
> [PAINTROID-703](https://catrobat.atlassian.net/browse/PAINTROID-703) and [PAINTROID-455](https://catrobat.atlassian.net/browse/PAINTROID-455) are also facing some issues regarding the BottomSheet, maybe its worth to discuss how we can deal with BottomSheets in genreal in the project @bakicelebi @juliajulie95 @exland @bhav-khurana ?

If we can set constraints in there I suggest to create a custom ThemeData: 

1. create an abstract class PaintroidThemeData
2. create LightPaintroidThemeData that extends PaintroidThemeData so we can also support darkmode in the future
   - in here you define the custom ThemeData and you can also adjust the BottomSheetWidth.

That is how I have done it in previous projects. I have code for this so I can set this up :)

**Comment by juliajulie95** [comment]
@msesko Setting up ThemeData sounds good to me! 👍

**Comment by Lenkomotive** [comment]
> @msesko Setting up ThemeData sounds good to me! 👍

I agree. 👍🏼 This will give us more flexibility in the future dealing with anything related.

**Comment by msesko** [comment]
> > @msesko Setting up ThemeData sounds good to me! 👍
> 
> I agree. 👍🏼 This will give us more flexibility in the future dealing with anything related.

Yes if you look at the code the bottomSheet is already fixed. I just need to match the design to look exactly the same as before.

File: packages/component_library/lib/src/theme/data/paintroid_theme_data.dart [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import 'package:component_library/component_library.dart';
+import 'package:flutter/material.dart';
+
+const dividerThemeData = DividerThemeData(
```
Comment (Lenkomotive) [inline packages/component_library/lib/src/theme/data/paintroid_theme_data.dart]: Why not include this into the abstract calls below?

File: packages/component_library/lib/src/theme/data/paintroid_theme_data.dart [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  space: 0,
+);
+
+ElevatedButtonThemeData get buttonThemeData => ElevatedButtonThemeData(
```
Comment (Lenkomotive) [inline packages/component_library/lib/src/theme/data/paintroid_theme_data.dart]: this too...

File: packages/component_library/lib/src/theme/state/theme_mode_state_notifier.dart
Comment (Lenkomotive) [inline packages/component_library/lib/src/theme/state/theme_mode_state_notifier.dart:1]: Please use generated Riverpod

File: packages/component_library/lib/src/theme/theme_provider.dart
Comment (Lenkomotive) [inline packages/component_library/lib/src/theme/theme_provider.dart:1]: please use generated Riverpod

File: packages/features/onboarding_screen/lib/src/screens/screen2.dart [hunk 3]
```diff
@@ -78,6 +47,24 @@ class _Screen2State extends State<Screen2> {
 
   @override
   Widget build(BuildContext context) {
+    var title = Text(
+      titleText,
+      style: TextStyle(
```
Comment (Lenkomotive) [inline packages/features/onboarding_screen/lib/src/screens/screen2.dart]: Consider defining TextStyles for the screens

**Review by Lenkomotive — CHANGES_REQUESTED** [review]

**Review by Lenkomotive — APPROVED** [review]
👍🏼

**Pull request merged**